From 8aaf4cc32032177404808bf84dc88b53d27b75aa Mon Sep 17 00:00:00 2001
From: Tzvetomir Stoyanov <tz.stoyanov@gmail.com>
Date: Mon, 18 Aug 2025 05:51:50 +0300
Subject: [PATCH 2/2] LFS: Add support for Pico multicore.

When Pico is used in multicore mode, LFS must run in thread safe mode to
protect its logic.

Signed-off-by: Tzvetomir Stoyanov <tz.stoyanov@gmail.com>
---
 CMakeLists.txt | 3 +++
 lfs_pico.h     | 8 ++++++++
 2 files changed, 11 insertions(+)
 create mode 100644 lfs_pico.h

diff --git a/libs/littlefs-lib/CMakeLists.txt b/libs/littlefs-lib/CMakeLists.txt
index d085f2c..e6cc501 100644
--- a/libs/littlefs-lib/CMakeLists.txt
+++ b/libs/littlefs-lib/CMakeLists.txt
@@ -1,6 +1,9 @@
 add_library(littlefs-lib INTERFACE)
 target_include_directories(littlefs-lib INTERFACE ${CMAKE_CURRENT_LIST_DIR})
 target_link_libraries(littlefs-lib INTERFACE hardware_flash)
+
+target_compile_options(littlefs-lib INTERFACE -DLFS_DEFINES=lfs_pico.h)
+
 target_sources(littlefs-lib INTERFACE
     ${CMAKE_CURRENT_LIST_DIR}/lfs.c
     ${CMAKE_CURRENT_LIST_DIR}/lfs_util.c
diff --git a/libs/littlefs-lib/lfs_pico.h b/libs/littlefs-lib/lfs_pico.h
new file mode 100644
index 0000000..0570d00
--- /dev/null
+++ b/libs/littlefs-lib/lfs_pico.h
@@ -0,0 +1,8 @@
+#ifndef __LFS_PICO_H__
+#define __LFS_PICO_H__
+
+#if LIB_PICO_MULTICORE
+#define LFS_THREADSAFE
+#endif
+
+#endif /* __LFS_PICO_H__ */
\ No newline at end of file
-- 
2.48.1

